scripts/test_lib.sh

```diff
@@ -236,11 +269,33 @@ function go_test {
   local packages="${1}"
   local mode="${2}"
   local flags_for_package_func="${3}"
+  local junit_filename_prefix
 
   shift 3
 
   local goTestFlags=""
   local goTestEnv=""
+
+  ##### Create a junit-style XML test report in this directory if set. #####
+  JUNIT_REPORT_DIR=${JUNIT_REPORT_DIR:-}
+
+  # If JUNIT_REPORT_DIR is unset, and ARTIFACTS is set, then have them match.
+  if [[ -z "${JUNIT_REPORT_DIR:-}" && -n "${ARTIFACTS:-}" ]]; then
+    export JUNIT_REPORT_DIR="${ARTIFACTS}"
+  fi
+
+  # Used to filter verbose test output.
+  go_test_grep_pattern=".*"
+
+  if [[ -n "${JUNIT_REPORT_DIR}" ]] ; then
+    goTestFlags+="-v "
+    goTestFlags+="-json "
+    # Show only summary lines by matching lines like "status package/test"
+    go_test_grep_pattern="^[^[:space:]]\+[[:space:]]\+[^[:space:]]\+/[^[[:space:]]\+"
```

Comment:
Does it mean that if the test fails, we will still see only summary without integration with the 'xml' viewer ?

Reply:
Could you, please, generate a run with artificially failing test to take a look how the results (logs) from the failed test-case would look.

Reply:
@ptabor The `go_test_grep_pattern` is to filter the `go test` output in the usual log/console and display only the Summary lines from `json` output when JUNIT Reporting is enabled.
When JUNIT Reporting is NOT enabled, the `go_test_grep_pattern` is ".*" thus displaying the complete output of `go test`.
Due to this `go_test_grep_pattern` when JUNIT is enabled, we get below kind of output:
![image](https://user-images.githubusercontent.com/63038004/123312325-d05abf80-d545-11eb-947f-e16cb60f3384.png)

Reply:
> 
> 
> Could you, please, generate a run with artificially failing test to take a look how the results (logs) from the failed test-case would look.

We have a failing test on our ppc64le architecture. Can you please tell me if the below link is accessible?
https://prow.ppc64le-cloud.org/view/s3/prow-logs/logs/postsubmit-master-golang-etcd-build-test-ppc64le/1404900977844162560

Reply:
thx. Seems good.